contests/_36-PDP/blyk-publib-statement.md

```diff
@@ -0,0 +1,88 @@
+---
+layout: statement
+codename: publib
+---
+
+Δίνεται ο οδικός χάρτης ενός απομακρυσμένου νησιού της χώρας. Στο
+νησί υπάρχουν $$N$$ χωριά (αριθμημένα από $$1$$ μέχρι $$N$$) που συνδέονται
+μεταξύ τους με $$M$$ δρόμους, καθένας από τους οποίους είναι διπλής
+κατεύθυνσης και συνδέει μεταξύ τους δύο χωριά. Θεωρήστε δεδομένο
+ότι είναι δυνατόν να μεταβεί κανείς από οποιοδήποτε χωριό του νησιού
+σε οποιοδήποτε άλλο, μέσω των δρόμων του οδικού δικτύου.
+
+Στο νησί μέχρι σήμερα δεν υπάρχει δημόσια βιβλιοθήκη. Ιδανικά, κάθε
+χωριό θα έπρεπε να έχει τη δική του. Όμως τα χρήματα του δημόσιου
+ταμείου φτάνουν για μία μόνο δημόσια βιβλιοθήκη, που θα πρέπει
+αναγκαστικά θα τοποθετηθεί σε ένα από τα χωριά του νησιού.
+
+Ονομάζουμε “απόσταση” ενός χωριού από τη βιβλιοθήκη το ελάχιστο
+πλήθος των δρόμων που πρέπει να διανύσουμε για να φτάσουμε από
+το χωριό σε αυτήν (δηλαδή, το πλήθος των ακμών στο συντομότερο
+μονοπάτι). Για παράδειγμα, αν η βιβλιοθήκη τοποθετηθεί στο χωριό 1,
+τότε το χωριό 1 βρίσκεται σε απόσταση 0 από τη βιβλιοθήκη. Αν το
+χωριό 2 είναι γειτονικό του 1 (δηλαδή συνδέεται απευθείας με αυτό με
+έναν δρόμο), τότε το χωριό 2 βρίσκεται σε απόσταση 1 από τη
+βιβλιοθήκη. Αν το χωριό 3 είναι γειτονικό του 2 αλλά όχι γειτονικό του
+1 (δηλαδή συνδέεται απευθείας με το χωριό 2 αλλά όχι με το χωριό
+1), τότε το χωριό 3 βρίσκεται σε απόσταση 2 από τη βιβλιοθήκη. Και
+ούτω καθεξής.
+
+Θέλουμε να τοποθετήσουμε τη βιβλιοθήκη έτσι ώστε η απόσταση του
+πιο μακρινού χωριού από αυτήν να είναι η ελάχιστη δυνατή.
+
+
+## Πρόβλημα
+
+Να αναπτύξετε ένα πρόγραμμα σε μια από τις γλώσσες του IOI
+(PASCAL, C, C++, Java) το οποίο θα διαβάζει τον οδικό χάρτη του
+νησιού και θα εκτυπώνει την απόσταση του πιο μακρινού χωριού από
+τη βιβλιοθήκη, αν τοποθετήσουμε αυτή στην καλύτερη δυνατή θέση.
+
+
+## Αρχεία εισόδου:
+
+Τα αρχεία εισόδου με όνομα **publib.in** είναι αρχεία κειμένου με την
+εξής δομή: Στην πρώτη γραμμή υπάρχουν δύο θετικοί ακέραιοι
```

Comment:
Η εκφώνηση της ΕΕ δεν είναι όμορφα εκφρασμένη (έχει δυο ':' στην ίδια πρόταση). Μήπως να το αλλάζαμε σε:

Τα αρχεία εισόδου με όνομα **publib.in** είναι αρχεία κειμένου με την
εξής δομή:< br >
Στην πρώτη γραμμή υπάρχουν δύο θετικοί ακέραιοι
αριθμοί χωρισμένοι μεταξύ τους με ένα κενό διάστημα οι οποίοι αναπαριστούν το πλήθος των
χωριών $$N$$ και το πλήθος των δρόμων $$M$$ που τα συνδέουν

Reply:
Άλλαξα λίγο την διατύπωση